pkg/api/openshiftcluster.go

```diff
@@ -204,4 +204,6 @@ const (
 	InstallPhaseDeployStorage InstallPhase = iota
 	InstallPhaseDeployResources
 	InstallPhaseRemoveBootstrap
+	InstallPhaseSetupRouter
```

Comment:
I am worried about adding more phases because I think they could easily play badly with RP upgrades.  If we have installs running in flight as we upgrade, and we add/remove phases, things could get pretty weird.

I wonder whether we should be trying to reduce to two phases - "everything before we delete the bootstrap", and "bootstrap deletion and everything after".

I'm not against the division of the install process into smaller functions or separate files.  My inclination here is to not add new phases and to open an issue to condense the phases that we do have.  WDYT?

Reply:
> I wanted to have more steps so it's easier to understand where we failed during installation without looking at logs. It might also be useful if we decide to gather stats to make decisions on areas where we need to improve, etc

I agree with this, I think we should aim to achieve it outside of the phase mechanism.  Could I ask you to open two issues - one to get the number of phases down, the second to implement something to address ^

Reply:
Created https://github.com/Azure/ARO-RP/issues/106 and https://github.com/Azure/ARO-RP/issues/107